#!/usr/bin/env python

PACKAGE="cubeeye_camera_node"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#             Name                               Type       Level  Description                                     Default    Min       Max
gen.add("amplitude_threshold_max",               int_t,     1,     "amplitude threshold max",                      65535,         0,    65535)
gen.add("amplitude_threshold_min",               int_t,     2,     "amplitude threshold min",                      0,         0,        255)
gen.add("auto_exposure",                         bool_t,    3,     "auto exposure on/off",                         True)
gen.add("depth_range_min",                       int_t,     4,     "depth range min",                              150,       0,        65535)
gen.add("depth_range_max",                       int_t,     5,     "depth range max",                              65535,     0,        65535)
gen.add("flying_pixel_remove_filter",            bool_t,    6,     "flying pixel remove filter on/off",            True)
gen.add("flying_pixel_remove_threshold",         int_t,     7,     "flying pixel remove filter threshold",         3000,      0,        65535)
gen.add("framerate",                             int_t,     8,     "framerate",                                    15,        5,        15)
gen.add("integration_time",                      int_t,     9,     "integration time (Depends on auto_exposure)",  1000,      0,        2000)
gen.add("median_filter",                         bool_t,    10,    "median filter",                                False)
gen.add("noise_removal_threshold",               int_t,     11,    "noise removal threshold",                      700,       0,        65535)
gen.add("outlier_remove_filter",                 bool_t,    12,    "outlier removal filter on/off",                True)
gen.add("phase_noise_filter",                    bool_t,    13,    "outlier removal filter on/off",                False)
gen.add("scattering_filter",                     bool_t,    14,    "scattering removal filter on/off",             False)
gen.add("scattering_filter_threshold",           int_t,     15,    "scattering removal threshold",                 700,       0,        65535)

exit(gen.generate(PACKAGE, "cubeeye_camera_node", "cubeeye_icube"))

